core/src/main/java/kafka/server/share/SharePartition.java

```diff
@@ -547,12 +547,136 @@ public CompletableFuture<Optional<Throwable>> acknowledge(
+                InFlightBatch inFlightBatch = entry.getValue();
+
+                if (inFlightBatch.offsetState() == null
+                        && inFlightBatch.batchState() == RecordState.ACQUIRED
+                        && inFlightBatch.batchMemberId().equals(memberId)
+                        && checkForStartOffsetWithinBatch(inFlightBatch.firstOffset(), inFlightBatch.lastOffset())) {
+                    // For the case when batch.firstOffset < start offset <= batch.lastOffset, we will be having some
+                    // acquired records that need to move to archived state despite their delivery count.
+                    inFlightBatch.maybeInitializeOffsetStateUpdate();
+                }
+
+                if (inFlightBatch.offsetState() != null) {
+                    throwable = releaseAcquiredRecordsForPerOffsetBatch(memberId, inFlightBatch, recordState, updatedStates, stateBatches);
+                    if (throwable != null)
+                        break;
+                    continue;
+                }
+                throwable = releaseAcquiredRecordsForCompleteBatch(memberId, inFlightBatch, recordState, updatedStates, stateBatches);
+                if (throwable != null) {
+                    break;
+                }
+            }
+
+            // If the release acquired records is successful then persist state, complete the state transition
+            // and update the cached state for start offset. Else rollback the state transition.
+            rollbackOrProcessStateUpdates(throwable, updatedStates, stateBatches);
+        } finally {
+            lock.writeLock().unlock();
+        }
+        return CompletableFuture.completedFuture(Optional.ofNullable(throwable));
+    }
+
+    private Throwable releaseAcquiredRecordsForPerOffsetBatch(String memberId,
+                                                              InFlightBatch inFlightBatch,
+                                                              RecordState recordState,
+                                                              List<InFlightState> updatedStates,
+                                                              List<PersisterStateBatch> stateBatches) {
+
+        log.trace("Offset tracked batch record found, batch: {} for the share partition: {}-{}", inFlightBatch,
+                groupId, topicIdPartition);
+        for (Map.Entry<Long, InFlightState> offsetState : inFlightBatch.offsetState.entrySet()) {
+
+            // Check if member id is the owner of the offset.
+            if (!offsetState.getValue().memberId().equals(memberId) && !offsetState.getValue().memberId().equals(EMPTY_MEMBER_ID)) {
+                log.debug("Member {} is not the owner of offset: {} in batch: {} for the share"
+                        + " partition: {}-{}. Skipping offset.", memberId, offsetState.getKey(), inFlightBatch, groupId, topicIdPartition);
+                return null;
```

Comment:
Shall we be consistent with other methods where it returns `Optional<Throwable>` instead of Throwable and null handling?